pkg/cmd/server.go

```diff
@@ -120,9 +170,119 @@ func newLogger(opts *serverOpts) *logrus.Logger {
 		logger.Level = logrus.InfoLevel
 	} else {
 		logger.Level = lvl
+		logger.Infof("Setting log-level=%s", lvl.String())
 	}
 
 	logger.Formatter = new(logrus.JSONFormatter)
 
 	return logger
 }
+
+// newServer return server
+func newServer(cfg Config, opts *serverOpts, logger *logrus.Logger) (*server, error) {
+	// update qps and burst to config before accessing clientset
+	if err := cfg.SetClientQPS(opts.clientQPS); err != nil {
+		return nil, err
+	}
+
+	if err := cfg.SetClientBurst(opts.clientBurst); err != nil {
+		return nil, err
+	}
+
+	kubeClient, err := cfg.KubeClient()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to fetch kubernetes client")
+	}
+
+	dmaasClient, err := cfg.Client()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to fetch dmaas client")
+	}
+
+	veleroClient, err := cfg.VeleroClient()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to fetch velero client")
+	}
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+
+	return &server{
+		namespace:             cfg.GetNamespace(),
+		openebsNamespace:      opts.openebsNamespace,
+		veleroNamespace:       opts.veleroNamespace,
+		kubeClient:            kubeClient,
+		dmaasClient:           dmaasClient,
+		veleroClient:          veleroClient,
+		sharedInformerFactory: informers.NewSharedInformerFactoryWithOptions(dmaasClient, 0, informers.WithNamespace(cfg.GetNamespace())),
+		ctx:                   ctx,
+		logger:                logger,
+		cancelFunc:            cancelFunc,
+		clock:                 clock.RealClock{},
+		opts:                  opts,
+	}, nil
+}
+
+var (
+	// defaultControllerWorker is default value of controller worker
+	defaultControllerWorker int = 1
+)
+
+// listController returns list of supported controller
```

Comment:
listController sounds like 'get the existing ones'.. but here new ones are created..
can we change to 'getSupportedControllers'?

Reply:
ok